source/CardGames.py

```diff
@@ -270,50 +441,83 @@ def index(self, player_wanted, info):
 
   def surrender_card(self):
     temp_list1 = []
-    temp_list2 = []
     counter = 0
+    self.count_cards()
+    if len(self.info_player_asked) == 0:
+          self.game_over()
+
     for i in self.info_player_asked:
-      print(str(i), i.suit)
-      
       if str(i) != str(self.value_wanted):
         counter += 1 
-      elif i.suit != str(self.suit_wanted):
-        counter += 1
       else:
         temp_list1.append(self.info_player_asked[counter])
-        self.player_turn.addCard(temp_list1[0])
-        
-        for i in range(len(self.info_player_asked)):
-          card = self.info_player_asked.pop()
-          if str(card) != str(self.value_wanted):
-            temp_list2.append(card)
-          elif card.suit != str(self.suit_wanted):
-            temp_list2.append(card)
-
-        for i in range(len(temp_list2)):
-          self.info_player_asked.append(temp_list2[i])
-
-    print(self.player_turn)
-    self.player_turn.showHand()
-    print(self.player_asked)
-    self.player_asked.showHand()
+        counter += 1 
 
+    if len(temp_list1) != 0:
+      for card in temp_list1:
+        self.player_turn.addCard(card)
+        self.info_player_asked.remove(card)
+        
 
-  def ask_card(self):
-    self.player_deck_info()
-    player_turn = input(str("What is your name? "))
-    player_asked = input(str("Who would like to ask? "))
+    for player in self.players:
+      print(str(player))
+      player.showHand()
+      print()
 
-    self.value_wanted = int(input("What value would you like? "))
-    self.suit_wanted = str(input("What suit would you like? "))
 
-    self.info_player_turn = self.playerinfo[self.index(player_turn, self.playerinfo)][1]
-    self.info_player_asked = self.playerinfo[self.index(player_asked, self.playerinfo)][1]
+  def sayGoFish(self):   
+    width = os.get_terminal_size().columns
+    print()
+    deco = "*-"*(width//2)
+    response = input("Do you have any of the requested cards? (YES/NO): ")      
+    if response.upper() == "NO": 
+      self.player_turn.addCard(self.deck.getCard())
+            
+      print()
+      return print("GO FISH!!".center(width), deco)
+      
+    elif response.upper() == "YES":          
+      return "go_again"
+    else:          
+      return "Looks like there was a typo in your response, please try again"
+#self.surrender_card()
+
+  def playerages(self):
+      playerAge = []
+      for player in self.players:
+          while True:
+              try:
+                  age = int(input("Enter {}'s age: ".format(player.name)))
+                  playerAge.append((player, age))
+                  break
+              except ValueError:
+                  print("Please enter a valid age: ")
+      playerAge = sorted(playerAge, key=lambda x: x[1])
+      youngestPlayer = playerAge[0][0]
+      print("{} goes first!\n".format(youngestPlayer.name))
+      return youngestPlayer
+
```

Comment:
sayGoFish and playerages are awesome! Thank you for adjusting and adding these functions Keren!

Reply:
Of course!